src/components/pdf-highlighter/utils/rects.js

```diff
@@ -0,0 +1,81 @@
+import { sortBy } from 'lodash'
+
+// TODO: allow to highlight when PDF is rotated
+const normalizeToScale1 = ({ top, left, width, height }, scale) => {
+  if (scale >= 1) {
+    return {
+      top: top / scale,
+      left: left / scale,
+      width: width / scale,
+      height: height / scale,
+    }
+  }
+  return {
+    top: top * scale,
+    left: left * scale,
+    width: width * scale,
+    height: height * scale,
+  }
+}
+
+export const groupRectsByPage = (rects, pagesRange, pdfViewer) => {
+  let pageNode = pagesRange.selectionStartPage.node
+  let pageRect = pageNode.getBoundingClientRect()
+
+  const shiftPage = pageRect.top
+
+  // seems like it's already sorted but I couldn't verify it in API doc
+  // https://developer.mozilla.org/en-US/docs/Web/API/Element/getClientRects
+  const sortedRects = sortBy(Array.from(rects), [r => r.top])
```

Comment:
Use _Array.prototype_'s methods where it's possible

Reply:
https://github.com/oacore/reader/pull/34#discussion_r329319856